### PR TITLE
Add changes from rector rule ChangeOrIfContinueToMultiContinueRector

### PR DESCRIPTION
**Pull request by Harfusha**
https://github.com/cakephp/cakephp/issues/17756

### CODE REVIEW

**Comment by ADmad** [comment]
I don't see why the `||` need to be split into separate `if` blocks. Seems unnecessarily opinionated to me.

File: src/Core/PluginCollection.php
Comment (Harfusha) [inline src/Core/PluginCollection.php:1]: I think is is much more readable, because it splits the conditions that are not dependent on each other

**Comment by Harfusha** [comment]
I think the new syntax is much more readable, because conditions not dependent on each other are splited.

**Comment by Harfusha** [comment]
```
foreach ($cart as $key => $item) {
    if (intval($item['service_type_id']) !== SERVICE_TYPES_IDS_SALE && (empty($item['is_custom']) || ($item['price'] !== 0 && empty($item['price'])))) {
        $this->Cart->recalculatePrice($key, $cart);
    }
}
```

VS

```
foreach ($cart as $key => $item) {
    if (empty($item['service_type_id'])) {
        continue;
    }

    if ((int) $item['service_type_id'] === SERVICE_TYPES_IDS_SALE) {
        continue;
    }

    if (!empty($item['is_custom']) && ($item['price'] !== null || !empty($item['price']))) {
        continue;
    }

    $this->Cart->recalculatePrice($key, $cart);
}
```

I think the second one is much better (even when it has more functionality)

**Comment by ADmad** [comment]
For the above handpicked e.g. yes, but splitting code like `if ($connection === 'test' || $connection === 'default') {` into 2 `if` blocks is just silly.

**Comment by dereuromark** [comment]
I would also think there needs to be a threshold.

**Comment by Harfusha** [comment]
Concept is same :) I think the splitted syntax is generaly more readable.

In this simple condition i think it i readable in both ways.

The spliting will keep the readability of simple ones and improve readability of complex ones.

File: src/ORM/Marshaller.php [hunk 1]
```diff
@@ -585,25 +585,26 @@ public function merge(EntityInterface $entity, array $data, array $options = [])
...
+                if (
+                    $value === null
+                    && $original === $value
+                ) {
+                    continue;
+                }
+                if (
+                    is_object($value)
+                    && !($value instanceof EntityInterface)
+                    && $original == $value
```
Comment (ADmad) [inline src/ORM/Marshaller.php:607]: This is probably the only place where there's a reasonable readability improvement.

Comment (Harfusha) [reply src/ORM/Marshaller.php:607]: https://github.com/cakephp/cakephp/pull/17797/files#diff-a0632c31b68e91238cf1ebe123a4a11b3abc5c0fd7f6d0781891609a851cd4d7L55
https://github.com/cakephp/cakephp/pull/17797/files#diff-3daf27c5efc6d12787f801a3c83a33b29a25254c243c59e934913166a107a077L719

These are improved too in my case

Comment (ADmad) [reply src/ORM/Marshaller.php:607]: Please just keep this change or else close this PR.

**Comment by Harfusha** [comment]
Can't make change only on one place - will close it and add to ignore list

**Pull request closed without merging**